cli/crowdin.ts

```diff
@@ -135,95 +135,89 @@ function uploadBundledTranslationsAsync(crowdinDir: string, branch: string, prj:
-                        .filter(dir => !name || dir == "libs/" + name)
-                        .forEach(dir => {
-                            const locdir = path.join(dir, "_locales");
-                            if (fs.existsSync(locdir))
-                                fs.readdirSync(locdir)
-                                    .filter(f => /\.json$/i.test(f))
-                                    .forEach(f => files.push(path.join(locdir, f)))
-                        });
-                    return downloadFilesAsync(cred, files, "bundled");
-                });
+    if (!cred)
+        return;
+
+    await buildAllTranslationsAsync(async (fileName: string) => {
+        pxt.log(`downloading ${fileName}`);
+        return pxt.crowdin.downloadTranslationsAsync(cred.branch, cred.prj, cred.key, fileName, { translatedOnly: true, validatedOnly: true });
+    }, name);
+}
+
+export async function buildAllTranslationsAsync(langToStringsHandlerAsync: (fileName: string) => Promise<Map<Map<string>>>, singleDir?: string) {
+    await buildTranslationFilesAsync(["sim-strings.json"], "sim");
+    await buildTranslationFilesAsync(["target-strings.json"], "target");
+
+    const files: string[] = [];
+    pxt.appTarget.bundleddirs
+        .filter(dir => !singleDir || dir == "libs/" + singleDir)
+        .forEach(dir => {
+            const locdir = path.join(dir, "_locales");
+            if (fs.existsSync(locdir))
+                fs.readdirSync(locdir)
+                    .filter(f => /\.json$/i.test(f))
+                    .forEach(f => files.push(path.join(locdir, f)))
         });
 
-    function downloadFilesAsync(cred: CrowdinCredentials, todo: string[], outputName: string) {
-        const locs: pxt.Map<pxt.Map<string>> = {};
-        const nextFileAsync = (): Promise<void> => {
-            const f = todo.pop();
-            if (!f) {
-                return Promise.resolve();
-            }
+    await buildTranslationFilesAsync(files, "bundled");
 
+    async function buildTranslationFilesAsync(files: string[], outputName: string) {
+        const crowdinDir = pxt.appTarget.id;
+        const locs: pxt.Map<pxt.Map<string>> = {};
+        for (const filePath of files) {
             const errors: pxt.Map<number> = {};
-            const fn = path.basename(f);
+            const fn = path.basename(filePath);
             const crowdf = path.join(crowdinDir, fn);
-            const locdir = path.dirname(f);
+            const locdir = path.dirname(filePath);
             const projectdir = path.dirname(locdir);
-            pxt.log(`downloading ${crowdf}`);
-            pxt.debug(`projectdir: ${projectdir}`)
-            return pxt.crowdin.downloadTranslationsAsync(cred.branch, cred.prj, cred.key, crowdf, { translatedOnly: true, validatedOnly: true })
-                .then(data => {
-                    Object.keys(data)
-                        .filter(lang => Object.keys(data[lang]).some(k => !!data[lang][k]))
```

Comment:
dropped this filter as it's handled better by the `if (!dataLang || !stringifyTranslations(dataLang))` inside the foreach, which will return undefined if there aren't any non empty values